ddtrace/internal/hostname.py

```diff
@@ -11,3 +11,8 @@ def get_hostname():
     if not _hostname:
         _hostname = socket.gethostname()
     return _hostname
+
+
+def _reset():
```

Comment:
I wonder if we could make this a (auto) test fixture instead?

Reply:
I spoke to Brett about this. We considered that approach but we thought it would be best to add a function to access private global variable (_hostname). I could still add fixture which could wrap the hostname test with this reset function. 

Something like this maybe?

```
import pytest

@pytest.fixture(autouse=True)
def reset_hostname():
    hostname._reset()
    yield 
    hostname._reset()


@mock.patch("socket.gethostname")
def test_get_hostname(reset_hostname, socket_gethostname)
```

Reply:
@P403n1x87 were you thinking a global autouse fixture?

We figured since there is only a single test in this file for this method it shouldn't matter too much for now.